cdap-runtime-ext-remote-hadoop/src/main/java/io/cdap/cdap/runtime/spi/provisioner/remote/PingEdgeNodeCheck.java

```diff
@@ -0,0 +1,73 @@
+package io.cdap.cdap.runtime.spi.provisioner.remote;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class PingEdgeNodeCheck {
+  private static final Logger LOG = LoggerFactory.getLogger(PingEdgeNodeCheck.class);
+
+  public String selectPingableEdgeNode(List<String> hosts, int initialIndex, int timeout)
+    throws NoLiveEdgeNodeException {
+    int attemptsNumber = 0;
+    while (attemptsNumber < hosts.size()) {
+      String hostToCheck = hosts.get((initialIndex + attemptsNumber) % hosts.size());
+      String checkedHost = runNodePingCheck(hostToCheck, timeout);
+      if (checkedHost != null) {
+        return checkedHost;
+      }
+      attemptsNumber++;
+    }
+    throw new NoLiveEdgeNodeException(EdgeNodeCheckType.PING, hosts, timeout);
+  }
+
+  String runNodePingCheck(String hostToCheck, int timeout) {
+    try {
+      Process p = runNodePingCommand(hostToCheck);
+      boolean waitResult = p.waitFor(timeout, TimeUnit.MILLISECONDS);
```

Comment:
why we need this wait implementation?     ping command have `-w` argument for waiting for the response

see below the function `runPingCommand`. Make this one simpler with one place for debug section (like save the exception to variable and print it if the debug enabled? )


+ what is the sense to split the logic to 2 functions? should it be merged to one?  (runNodePingCheck + runNodePingCommand)

Reply:
`why we need this wait implementation? ping command have -w argument for waiting for the response` - we anyway need to have this functionality to guarantee the process won't block check execution.
`what is the sense to split the logic to 2 functions?` - for testing purpose